pkg/services/ngalert/api/api_ruler.go

```diff
@@ -196,6 +198,17 @@ func (srv RulerSrv) RoutePostNameRulesConfig(c *models.ReqContext, ruleGroupConf
 		return response.Error(http.StatusBadRequest, "rule group name is not valid", nil)
 	}
 
+	for _, r := range ruleGroupConfig.Rules {
+		cond := ngmodels.Condition{
+			Condition: r.GrafanaManagedAlert.Condition,
+			OrgID:     c.SignedInUser.OrgId,
+			Data:      r.GrafanaManagedAlert.Data,
+		}
+		if err := validateCondition(cond, c.SignedInUser, c.SkipCache, srv.DatasourceCache); err != nil {
+			return response.Error(http.StatusBadRequest, "failed to validate alert rule queries and expressions", err)
```

Comment:
Can we let the user known which rule and datasource are the ones that failed?

Reply:
the datasource name [is included in the error](https://github.com/grafana/grafana/blob/a54ac510c450748a816bf74ba52480e3671623d5/pkg/services/ngalert/api/util.go#L218) but I will add also the query refID.

Reply:
Ah, I see, is the format of the error that threw me off. Can we make it be

`failed to get datasource: data source 'unknown' not found"`

instead of

`failed to get datasource: unknown: data source not found"`

Reply:
on second thought: the `data source not found` part is the error returned from the cache. I would rather not changing it. WDYT?

Reply:
that's fair, we could simply do:

```
return nil, fmt.Errorf("%w: %s", err, datasourceUID)
```

To get: `data source not found: unknown"